packages/contracts-bedrock/scripts/ChainAssertions.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ProxyAdmin } from "src/universal/ProxyAdmin.sol";
+import { ResourceMetering } from "src/L1/ResourceMetering.sol";
+import { DeployConfig } from "scripts/DeployConfig.s.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { Constants } from "src/libraries/Constants.sol";
+import { L1StandardBridge } from "src/L1/L1StandardBridge.sol";
+import { L2OutputOracle } from "src/L1/L2OutputOracle.sol";
+import { ProtocolVersion, ProtocolVersions } from "src/L1/ProtocolVersions.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+import { L1CrossDomainMessenger } from "src/L1/L1CrossDomainMessenger.sol";
+import { OptimismMintableERC20Factory } from "src/universal/OptimismMintableERC20Factory.sol";
+import { L1ERC721Bridge } from "src/L1/L1ERC721Bridge.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { Types } from "scripts/Types.sol";
+import { Vm } from "forge-std/Vm.sol";
+
+library ChainAssertions {
+    function postDeployAssertions(
+        address systemConfig,
+        address protocolVersions,
+        DeployConfig cfg,
+        uint256 l2OutputOracleStartingTimestamp,
+        Vm vm
+    )
+        internal
+        view
+    {
+        SystemConfig sys = SystemConfig(systemConfig);
+        Types.ContractSet memory proxies = Types.ContractSet({
```

Comment:
This should be refactored to accept a contract set as an argument because it will be more flexible with existing + future deployments. This cannot run in simulation mode because the system config in prod doesn't have those getters. We can migrate to this style once this style is in production